Assets/Scripts/Audio/AudioManager.cs

```
@@ -0,0 +1,92 @@
+using UnityEngine;
+using FMODUnity;
+using FMOD.Studio;
+
+public enum AudioSurfaceType : int
+{
+    Concrete,
+    Wood,
+    Gravel,
+    Metal
+}
+
+public class AudioManager : MonoBehaviour
+{
+
+
+
+    public static AudioManager instance;
+
+    [SerializeField] private EventReference _testSound;
+
+    [SerializeField] private EventReference _pauseAudioEvent;
+    [SerializeField] private EventReference _pauseMenuSnapshot;
+
+    private const string _inGameSFXBusName = "bus:/SFX/In-World";
+    private const string _inGameUIBusName = "bus:/SFX/UI/In-World";
+    private const string _inGameVOBusName = "bus:/VO/In-World";
+    private const string _inGameMusicBusName = "bus:/Music/In-World";
+
+    public Bus _inGameSFXBus { get; private set; }
+    public Bus _inGameUIBus { get; private set; }
+    public Bus _inGameVOBus { get; private set; }
+    public Bus _inGameMusicBus { get; private set; }
+    private EventInstance _pauseMenuSnapshotInstance;
+
+    private void Awake()
+    {
+        instance = this;
```

We should have a check here to see if instance has been set or not. If it has, let's throw an exception and remove the component from its game object. I don't believe in being over-cautious with singletons.